cdq-analysis/pkg/componentdetectionquery.go

```diff
@@ -36,61 +38,144 @@ type K8sInfoClient struct {
-		Revision, err = GetBranchFromRepo(componentPath)
+	if revision == "" {
+		revision, err = GetBranchFromRepo(componentPath)
 		if err != nil {
-			log.Error(err, fmt.Sprintf("Unable to clone repo %s to path %s, exiting reconcile loop %v", URL, clonePath, namespace))
+			log.Error(err, fmt.Sprintf("Unable to get branch from cloned repo for component path %s, exiting reconcile loop %v", componentPath, namespace))
 			k.SendBackDetectionResult(devfilesMap, devfilesURLMap, dockerfileContextMap, componentPortsMap, name, namespace, err)
-			return nil, nil, nil, nil, "", err
+			return err
 		}
 	}
 
+	cdqInfo.ClonedRepo.ClonedPath = clonePath
+	cdqInfo.ClonedRepo.ComponentPath = componentPath
+	cdqInfo.GitURL.Revision = revision
+	return nil
+
+}
+
+// CDQ analyzer
+// return values are for testing purpose
+func CloneAndAnalyze(k K8sInfoClient, namespace, name, context string, cdqInfo *CDQInfoClient) (map[string][]byte, map[string]string, map[string]string, map[string][]int, string, error) {
+	log := k.Log
+	var clonePath, componentPath string
+	alizerClient := AlizerClient{}
+	devfilesMap := make(map[string][]byte)
+	devfilesURLMap := make(map[string]string)
+	dockerfileContextMap := make(map[string]string)
+	componentPortsMap := make(map[string][]int)
+	var Fs afero.Afero
+	var err error
+
+	var components []model.Component
+
+	repoURL := cdqInfo.GitURL.RepoURL
+	gitToken := cdqInfo.GitURL.Token
+	registryURL := cdqInfo.DevfileRegistryURL
+	devfilePath := cdqInfo.DevfilePath
+	dockerfilePath := cdqInfo.DockerfilePath
+
+	err = cdqInfo.clone(k, namespace, name, context)
+	if err != nil {
+		return nil, nil, nil, nil, "", err
+	}
+
+	// search the cloned repo for valid devfile locations
+	devfileBytes, err := FindValidDevfiles(cdqInfo)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to find from any known devfile locations from %s ", cdqInfo.GitURL))
+	}
+
+	// search the cloned repo for valid dockerfile locations
+	dockerfileBytes, err := FindValidDockerfile(cdqInfo)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to find from any known devfile locations from %s ", cdqInfo.GitURL))
+	}
+
+	isDevfilePresent := len(devfileBytes) != 0
+	isDockerfilePresent := len(dockerfileBytes) != 0
+
+	// the following values come from the previous step when the repo was cloned
+	Fs = cdqInfo.ClonedRepo.Fs
+	componentPath = cdqInfo.ClonedRepo.ComponentPath
+	clonePath = cdqInfo.ClonedRepo.ClonedPath
+	revision := cdqInfo.GitURL.Revision
+	devfilePath = cdqInfo.DevfilePath
+	dockerfilePath = cdqInfo.DockerfilePath
+
+	if context == "" {
+		context = "./"
+	}
+
+	//search for devfiles
+
 	isMultiComponent := false
 	if isDevfilePresent {
-		updatedLink, err := UpdateGitLink(URL, Revision, path.Join(context, devfilePath))
+		// devfilePath is the local path to the temp cloned dir.  We need to construct the git link with the devfile location only
+		devfileLoc := devfilePath[len(cdqInfo.ClonedRepo.ClonedPath)+1:] //account for the forward slash
```

Comment:
yeah, it's buried in the FindValidDevfile/Dockerfiles.  I'll need to update the usage since I assign the local, cloned temp path name for the devfilePath when it should be one of the four valid locations.